### PR TITLE
:broom: Add c10s and el10 repos, update distro aliases, update SPEC, c8s EOL

### PR DESCRIPTION
**Pull request by thozza**
Testing in c10s has been failing on a few things (at first on the non-existence of `podman-plugins`). While looking at composer, I realized that we never properly enabled c10s / el10, so I tried to address hopefully all missing pieces. And on top of that, I also extended our Packit config for c9s.

- Update the SPEC:
  - Don't require `podman-plugins` on F40+ and c10s / el10
  - Update which repo definitions are installed on RHEL / CS distros
- Packit:
  - Build c10s in COPR for all available arches
  - Build c9s in COPR for all available arches
- Add c10s and el10.0 repo definitions (official and testing)
- Update distro aliases in the config and RHOSP template.
- c8s EOL cleanup:
  - Removed COPR builds from Packit config.
  - Removed all c8s runners from GitLab CI config.
  - Removed c8s prod and testing repos and adjusted SPEC to not install them.

This pull request includes:

- [ ] adequate testing for the new functionality or fixed issue
- [ ] adequate documentation informing people about the change such as
  - [ ] submit a PR for the READMEs [listed here](https://github.com/osbuild/osbuild.github.io/blob/main/readme-list)
  - [ ] submit a PR for the [osbuild.org website](https://github.com/osbuild/osbuild.github.io/) repository if this PR changed any behavior not covered by the automatically updated READMEs

### CODE REVIEW

**Review by croissanne — DISMISSED** [review]

**Comment by thozza** [comment]
I also backported the PR to c10s to see if it helps with the CI tests - https://gitlab.com/redhat/centos-stream/rpms/osbuild-composer/-/merge_requests/111

**Comment by thozza** [comment]
> I also backported the PR to c10s to see if it helps with the CI tests - https://gitlab.com/redhat/centos-stream/rpms/osbuild-composer/-/merge_requests/111

c10s CI passed with this PR backported as a downstream patch.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -192,16 +192,17 @@ jobs:
...
-    name: "🔍 Check for valid snapshot urls"
-    runs-on: ubuntu-20.04
-    steps:
-      - name: Check out code into the Go module directory
-        uses: actions/checkout@v4
-        with:
-          ref: ${{ github.event.pull_request.head.sha }}
-      - name: Check for valid snapshot urls
-        run: ./tools/check-snapshots --errors-only .
+  # TODO: enable again once enumerating rpmrepo does not fail anymore
```
Comment (atodorov) [inline .github/workflows/tests.yml:195]: What do you mean by "fail anymore" ? 

I've seen this timing out on occasion but restarting the job usually works fine for me. Is there a more serious problem than that?

**Review by atodorov — CHANGES_REQUESTED** [review]

File: .github/workflows/tests.yml (hunk 1 above)
Comment (atodorov) [reply .github/workflows/tests.yml:195]: Also FTR TODO comments tend to get forgotten if we don't have tickets for them.

Comment (thozza) [reply .github/workflows/tests.yml:195]: We probably need to clean up some old snapshots, because the timeout limits are at the maximum.

Comment (thozza) [reply .github/workflows/tests.yml:195]: And this is now failing consistently also in the osbuild repo... restarting won't help.

Comment (achilleas-k) [reply .github/workflows/tests.yml:195]: I think it's failing more often than it's passing now.  It's getting increasingly unreliable and I've had occasions where I had to retry it 4-5 times to get it to pass.

**Review by achilleas-k — APPROVED** [review]
:heart: :tada:

**Review by schuellerf — APPROVED** [review]
LGTM

**Review by supakeen — APPROVED** [review]
**10/10**

File: .github/workflows/tests.yml (hunk 1 above)
Comment (thozza) [reply .github/workflows/tests.yml:195]: I created https://issues.redhat.com/browse/COMPOSER-2263

Comment (thozza) [reply .github/workflows/tests.yml:195]: @atodorov are you OK with keeping it as is with the ticket?

**Review by atodorov — APPROVED** [review]

**Pull request merged**